include/boundary_conditions.h

```diff
@@ -11,104 +11,73 @@
 #include <sys/time.h>
 #include <string.h>
 
-double Lx;
-double Ly;
-double Lz;
+#include <cmath>
+#include <stdexcept>
+#include <fstream>
+#include <ufc.h>
+
 
 /////////////////////////////////////////////////////////////////////////////
 ///  Sub Domains
 /////////////////////////////////////////////////////////////////////////////
 
-// Sub domain for x=-Lx and x=Lx boundary conditions
-// Before it was: class DirichletBoundary : public SubDomain
-class EastWestDomain: public SubDomain
-{
-    bool inside(const Array<double>& x, bool on_boundary) const
-    {
-        // Before it was: return on_boundary && (x[0] < -Lx+2.*DOLFIN_EPS or x[0] > Lx-2.*DOLFIN_EPS);
-        // Arthur changed it to:
-        return on_boundary && (x[0] < -Lx+DOLFIN_EPS || x[0] > Lx-DOLFIN_EPS); //(is that ok ?)
-    }
+class XBoundaries: public dolfin::SubDomain {
```

Comment:
I think we can simplify this by being more careful about how we setup the class.  I'll take a look at an implementation following [this post](http://stackoverflow.com/questions/120876/c-superclass-constructor-calling-rules)